### PR TITLE
Match PPG implementation

### PR DESCRIPTION
**Pull request by Dipamc77**
## Types of changes

- [ ] Bug fix
- [ ] New feature
- [x] New algorithm
- [ ] Documentation

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/vwxyzjn/cleanrl/blob/master/CONTRIBUTING.md) guide (**required**).
- [x] I have ensured `pre-commit run --all-files` passes (**required**).
- [x] I have updated the documentation and previewed the changes via `mkdocs serve`.
- [x] I have updated the tests accordingly (if applicable).

If you are adding new algorithms or your change could result in performance difference, you may need to (re-)run tracked experiments. See https://github.com/vwxyzjn/cleanrl/pull/137 as an example PR. 
- [x] I have contacted @vwxyzjn to obtain access to the [openrlbenchmark W&B team](https://wandb.ai/openrlbenchmark) (**required**).
- [x] I have tracked applicable experiments in [openrlbenchmark/cleanrl](https://wandb.ai/openrlbenchmark/cleanrl) with `--capture-video` flag toggled on (**required**).
- [x] I have added additional documentation and previewed the changes via `mkdocs serve`.
    - [x] I have explained note-worthy implementation details.
    - [x] I have explained the logged metrics.
    - [x] I have added links to the original paper and related papers (if applicable).
    - [x] I have added links to the PR related to the algorithm.
    - [x] I have created a table comparing my results against those from reputable sources (i.e., the original paper or other reference implementation).
    - [x] I have added the learning curves (in PNG format with `width=500` and `height=300`).
    - [x] I have added links to the tracked experiments.
- [x] I have updated the tests accordingly (if applicable).

### CODE REVIEW

**Review by vwxyzjn — COMMENTED** [review]
This is massive! Thanks for making this PR.

I think before any additional changes, our next step is to establish a great baseline. We should try to compare our results against the original results, otherwise, it's hard to attest the quality of our PPG implementation given that we included divergent changes such as using shared optimizers. 

Unfortunately, the original paper ran experiments using the `hard` distribution mode, so we might have to re-run them. I made a fork available here https://github.com/openrlbenchmark/phasic-policy-gradient to run tracked experiments. Unfortunately, I was not able to run them due to insufficient GPU memory... Would you mind giving it a try? The benchmark commands is at https://github.com/openrlbenchmark/phasic-policy-gradient/blob/add-wandb/benchmark.sh

File: docs/rl-algorithms/ppg.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+- The original PPG paper does not report results on easy environments, hence more hyperparameter tuning can give better results.
+- Skipping every alternate auxiliary phase gives similar performance on easy environments while saving compute.
+- Normalized network initialization scheme seems to matter a lot, but using layernorm with orthogonal initialization also works.
+- Using mixed precision for auxiliary phase also works well to save compute, but using on policy phase makes training unstable.
+
+
+### Differences from the original PPG code
+
+- The original PPG code supports LSTM whereas the CleanRL code does not.
+- The original PPG code uses separate optimizers for policy and auxiliary phase, but we do not implement this as we found it to not make too much difference.
```
[the remark this answers is not in the document]
Comment (Dipamc77) [reply docs/rl-algorithms/ppg.md]: Done

File: cleanrl/ppg_procgen.py [hunk 2]
```diff
@@ -82,32 +84,51 @@ def parse_args():
         help="E_aux:the K epochs to update the policy")
     parser.add_argument("--beta-clone", type=float, default=1.0,
         help="the behavior cloning coefficient")
-    parser.add_argument("--n-aux-minibatch", type=int, default=16 * 32,
+    parser.add_argument("--aux-num-rollouts", type=int, default=4,
```
[the remark this answers is not in the document]
Comment (Dipamc77) [reply cleanrl/ppg_procgen.py]: Done

File: cleanrl/ppg_procgen.py [hunk 3]
```diff
@@ -56,8 +56,10 @@ def parse_args():
         help="the lambda for the general advantage estimation")
     parser.add_argument("--num-minibatches", type=int, default=8,
         help="the number of mini-batches")
-    parser.add_argument("--norm-adv", type=lambda x: bool(strtobool(x)), default=True, nargs="?", const=True,
+    parser.add_argument("--norm-adv", type=lambda x: bool(strtobool(x)), default=False, nargs="?", const=True,
         help="Toggles advantages normalization")
+    parser.add_argument("--norm-adv-ppg", type=lambda x: bool(strtobool(x)), default=True, nargs="?", const=True,
```
[the remark this answers is not in the document]
Comment (Dipamc77) [reply cleanrl/ppg_procgen.py]: Changed to `--adv-norm-fullbatch`, wanted to avoid using `batch-norm` because common DL name.

**Comment by vwxyzjn** [comment]
Hey, @Dipamc77 I made some modifications to the documentation. Would you mind adding a section called "Explanation of the logged metrics"? (see [here](https://docs.cleanrl.dev/rl-algorithms/ppo/#explanation-of-the-logged-metrics) as an example). Everything else looks good on my end.

image

I did notice the wall-time performance of openai/phasic-policy-gradient is better, but the reason could be that we record the videos (which can be costly) or hardware differences. I am not too worried about it though.

image

**Review by vwxyzjn — APPROVED** [review]
